tools/jit/gen_jit_dispatch.py

```diff
@@ -149,18 +151,40 @@ def from_ivalue(arg, value):
 );
 """)
 CALL_NAMESPACE_WITH_TENSOR_OPTIONS = CodeTemplate("""\
+// This is a hack.
+// Please see [Add requires_grad to native_functions.yaml and potentially to TensorOptions object]
+// In the tracking issue: https://github.com/pytorch/pytorch/issues/30405
+const auto options = TensorOptions()
+        .dtype(${dtype})
+        .layout(${layout})
+        .device(${device})
+        .pinned_memory(${pin_memory});
+#ifdef USE_STATIC_DISPATCH
+    auto result_ = at::${name}(${args_with_tensor_options_disp});
+#else
+    auto result_ = torch::_${name}(${args_with_tensor_options});
```

Comment:
I'm not really sure how this variation of the template is related to the `requires_grad` hack

Reply:
Did you mean to say, only the `ARANGE` version of the template is the hack?